src/Microsoft.Health.SqlServer/Features/Storage/SqlTransactionHandler.cs

```diff
@@ -16,10 +16,9 @@ public class SqlTransactionHandler : ITransactionHandler
 
     public ITransactionScope BeginTransaction()
     {
-        Debug.Assert(SqlTransactionScope == null, "The existing SQL transaction scope should be completed before starting a new transaction.");
-
         if (SqlTransactionScope != null)
         {
+            Debug.Fail("The existing SQL transaction scope should be completed before starting a new transaction.");
```

Comment:
Should this be the message of the exception?

Reply:
Makes sense to me